src/Kestrel.Core/Internal/ConnectionHandler.cs

```diff
@@ -32,12 +32,14 @@ public void OnConnection(IFeatureCollection features)
         {
             var connectionContext = new DefaultConnectionContext(features);
 
-            var transportFeature = connectionContext.Features.Get<IConnectionTransportFeature>();
+            var applicationFeature = connectionContext.Features.Get<IApplicationTransportFeature>();
```

Comment:
We could expose these as first class properties on a KestrelConnection object but it's only used in this file. The HttpConnectionMiddleware is written against the `ConnectionContext` abstraction.